src/fsharp/FSharp.Core/prim-types.fsi

```diff
@@ -1930,6 +1930,12 @@ namespace Microsoft.FSharp.Collections
         /// <param name="endIndex">The end index.</param>
         /// <returns>The sub list specified by the input indices.</returns>
         member GetSlice : startIndex:int option * endIndex:int option -> 'T list  
+
+        /// <summary>Get the index corresponding to ^i.</summary>
+        /// <param name="dim">The dimension of the index.</param>
+        /// <param name="offset">The offset from the end.</param>
+        /// <returns>The corresponding index from the start.</returns>
+        member GetReverseIndex: dim: int * offset: int -> int
```

Comment:
This is a design question.

Do you think `GetIndexFromEnd` would be a better name? I think it's clearer on its intent. `GetReverseIndex` is ok; though, it means getting the index from the end with the specified rank and index.

Reply:
Other comments on design:

https://github.com/fsharp/fslang-suggestions/issues/358#issuecomment-553244981